module_utils/certificate/providers/certmonger.py

```diff
@@ -53,6 +53,35 @@ def __init__(self, *args, **kwargs):
         self._certmonger_dbus = CertmongerDBus()
         self._certmonger_metadata = self._get_certmonger_request()
 
+    def module_input_validation(self):
+        """Validate module input."""
+        principal = self.module.params.get("principal")
```

Comment:
`get("principal", [])` - I think this needs to return an empty list so the `for` loop below won't fail - `[]` will still evaluate to boolean `False`

Reply:
Fixed.